repo-agent/review-sandbox/main.go

```diff
@@ -401,3 +430,24 @@ func startCodeServer() (*exec.Cmd, error) {
 	log.Printf("Running code-server in subprocess %d\n", cmd.Process.Pid)
 	return cmd, nil
 }
+
+func dedupeAndCombineText(text string) (string, error) {
+	agentName := os.Getenv("AGENT_NAME")
+	provider, err := llm.NewLLMProvider(agentName)
+	if err != nil {
+		return "", err
+	}
+
+	if err := provider.Setup("/workspaces", "/tokens"); err != nil {
+		return "", err
+	}
+
+	prompt := fmt.Sprintf("The following text contains multiple sections separated by '---'. Please deduplicate and combine them into a single coherent section of text. Return only the result.\n\n%s", text)
```

Comment:
nit: can you add a comment here explaining where the sections are concatenated with `---`? IIUC it's from  the accumulatedAgentOutput (line 207) but might make sense to comment here for locality and also give semantic meaning to what is concatenated with ---

Reply:
added a comment in 2 places